provisioning/miniprov/hare_mp/cdf.py

```diff
@@ -64,11 +65,15 @@ def _get_cluster_id(self) -> str:
     def _create_node_descriptions(self) -> List[NodeDesc]:
         nodes: List[NodeDesc] = []
         conf = self.provider
+        services = conf.search_val('node', 'services',
+                                   Const.SERVICE_MOTR_IO.value)
+        services.extend(conf.search_val('node', 'services',
+                                        Const.SERVICE_S3_SERVER.value))
         machines: Dict[str, Any] = conf.get('node')
         for machine_id in machines.keys():
-            node_type = conf.get(f'node>{machine_id}>type')
-            # Skipping for controller node
-            if node_type == 'storage_node':
+            result = list(filter(lambda svc: machine_id in svc, services))
```

Comment:
Can you give a try list comprehensions? I think in your case the code will be more readable.
```suggestion
            result = [x for x in services if machine_id in x]
```